wasted-app/RestaurantRegistrationControl.cs

```diff
@@ -60,29 +61,16 @@ private void RegisterButton_Click(object sender, EventArgs e)
             if (passwordTextBox.Text == repeatPasswordTextBox.Text && CheckIfTextBoxesAreFull())
             {
                 ServicesController controller = ServicesController.Instance;
-                var mail = mailTextBox.Text;
-                var password = passwordTextBox.Text;
-                var restaurant = controller.RestaurantService.GetByMail(new Mail(mail));
-                if (restaurant == null)
+                try 
                 {
-                    var credentialError = GetValidationError(mail, password);
-                    if (credentialError == "")
-                    {
-                        passwordError.Text = "";
-                        Credentials creds = new Credentials(mail, password);
-                        controller.RestaurantService.Register(creds, new Restaurant("todo", restaurantNameTextBox.Text, new Coords(Convert.ToDecimal(latitudeTextBox.Text), Convert.ToDecimal(longitudeTextBox.Text)), new Credentials()));
-                        MessageBox.Show("Registered successfully");
-                        GoBack();
-                        ResetTextBoxes();
-                    }
-                    else
-                    {
-                        passwordError.Text = credentialError;
-                    }
+                    controller.RestaurantService.Register(new Credentials(mailTextBox.Text, passwordTextBox.Text), new Restaurant("todo", restaurantNameTextBox.Text, new Coords(Convert.ToDecimal(latitudeTextBox.Text), Convert.ToDecimal(longitudeTextBox.Text)), new Credentials()));
```

Comment:
This is quite unreadable in my opinion. Maybe extract arguments to variables or at least put arguments on different lines.

Reply:
I'd say better to move into separate variables

Reply:
Fixed it.